web_src/js/features/formatting.js

```diff
@@ -11,4 +11,11 @@ export function initFormattingReplacements() {
       el.textContent = formatted;
     }
   }
+
+  // for each <time></time> tag, if it has the data-format="date" attribute, format
+  // the text according to the user's chosen locale
+  const formatter = new Intl.DateTimeFormat(lang, {year: 'numeric', month: 'long', day: 'numeric'});
```

Comment:
One more small optimization: Move `formatter` out of the function to line 4 and rename it to `dateFormatter`.